sig-scalability/slos/slos.md

```diff
@@ -0,0 +1,108 @@
+# Kubernetes scalability and performance SLIs/SLOs
+
+## What Kubernetes guarantees?
+
+One of the important aspects of Kubernetes is its scalability and performance
+characteristic. As Kubernetes user or operator/administrator of a cluster
+you would expect to have some guarantees in those areas.
+
+The goal of this doc is to organize the guarantees that Kubernetes provides
+in these areas.
+
+## What do we require from SLIs/SLOs?
+
+We are going to define more SLIs and SLOs based on the most important indicators
+in the system.
+
+Our SLOs need to have the following properties:
+- <b> They need to be testable </b> <br/>
+  That means that iwe need to have a benchmark to measure it it's met.
+- <b> They need to be understandable for users </b> <br/>
+  In particular, they need to be understandable for people not familiar
+  with the system internals, i.e. their formulation can't depend on some
+  arcane knowledge.
+
+However, we may introduce some internal (for developers only) SLIs, that
+may be useful for understanding performance characterstic of the system,
+but for which we don't provide any guarantees for users and thus may not
+be fully understandable for users.
+
+On the other hand, we do NOT require that our SLOs:
+- are measurable in a running cluster (though that's desired if possible) <br/>
+  In other words, not SLOs need to be easily translatable to SLAs.
+  Being able to benchmark is enough for us.
+
+## Types of SLOs
+
+While SLIs are very generic and don't really depend on anything (they just
+define what and how we measure), it's not the case for SLOs.
+SLOs provide guarantees, and satisfying them may depend on meeting some
+specific requirements.
+
+As a result, we build our SLOs in "you promise, we promise" format.
+That means, that we provide you a guarantee only if you satisfy the requirement
+that we put on you.
+
+As a consequence we introduce the two types of SLOs.
+
+### Steady state SLOs
+
+With steady state SLOs, we provide guarantees about system's behavior during
+normal operations. We are able to provide much more guarantees in that situation.
+
+```Definition
+We define system to be in steady state when the cluster churn per second is <= 20, where
```

Comment:
So we want to use the same number (20) for all sizes of clusters? As large clusters IIUC would expectedly have more churn, no?

Reply:
Longer term - maybe. For now, we don't really support bigger churn for larger clusters.